Source/Csla/Core/BusinessBase.cs

```diff
@@ -3018,9 +3018,11 @@ protected void LoadPropertyAsync<R>(PropertyInfo<R> property, Task<R> factory)
 
     #region IsBusy / IsIdle
 
-    [NonSerialized]
-    [NotUndoable]
-    private bool _isBusy;
+    private int _isBusyCounter;
+    /// <summary>
+    /// Indicates object is busy or not 
+    /// </summary>
+    public bool _isBusy => _isBusyCounter > 0;
```

Comment:
This implementation has a couple of issues:
1. It should _not_ introduce a new property but replace the old bool field with a counter.
2. The MarkBusy/Idle methods do not work in a parallel scenario because you are comparing the field and not the returned value from  `Interlocked`.
3. There are more objects which havca `Is(Self)Busy` implementation which should also be changed

Reply:
1. It should not introduce a new property but replace the old bool field with a counter. - **Have changed**. 

2. The MarkBusy/Idle methods do not work in a parallel scenario because you are comparing the field and not the returned value from Interlocked. - **Have changed**. 

3. There are more objects which have Is(Self)Busy implementation which should also be changed. - I have changed in `BusinessBase.cs` and `ReadOnlyBase.cs`.

I have found Is(Self)Busy fields in other class `ObservableBindingList.cs` and `ExtendedBindingList.cs` but, there 

```
 public virtual bool IsBusy
 {
   get { return false; }
 }
public virtual bool IsSelfBusy
{
  get { return IsBusy; }
}
```
------------------------------
```
public virtual bool IsBusy
{
  get { throw new NotImplementedException(); }
}
public virtual bool IsSelfBusy
{
  get { return IsBusy; }
}
```

Do I need to change Logic in these class? if yes then I think I have to create sub milestone. let me know if i am missing any classes.